Initial mapping for IHCs and SGNs

schilling40 opened the pull request:
The goal of this branch is to introduce the tonotopic mapping of SGNs and IHCs. The cells are mapped onto the length of the cochlea's spiral structure. This is achieved by determining the shortest path between the two most distant nodes within the cochlear component of the segmentation.
More intricate mappings, such as mapping the centrality of SGNs, have yet to be implemented.

WIP

constantinpape commented on flamingo_tools/segmentation/cochlea_mapping.py:
```diff
@@ -0,0 +1,141 @@
+    min_component_length: int = 50,
+    cell_type: str = "ihc",
+    filter_factor: Optional[float] = None
+) -> pd.DataFrame:
```
In addition, this function should also compute the "distance to path" for each node, with distance to center = 0 encoding nodes on the paths. We need this for example when sampling locations along the run length of the cochlea, where we want to sample positions from the center.

schilling40 commented on flamingo_tools/segmentation/cochlea_mapping.py:
```diff
@@ -0,0 +1,141 @@
+    max_dist = 0
+    farthest_pair = (None, None)
+
+    for u, dist_dict in all_lengths.items():
```
The bottleneck of this function is `nx.all_pairs_dijkstra_path_length(G, weight=weight)`, so I will try to find a faster alternative if I have time.

schilling40 commented:
The last commit addressed most of the comments by creating separate functions to measure the run length of the cochlea for SGNs and IHCs , by using "spatially stratified" subsampling with a voxel grid, and by creating a function to assign frequency ranges based on tonotopic mapping.
However, the original function has not yet been fully refactored, as much of the code overlaps after computing a central path.
Final adjustments will be made once the best possible method for determining a central path for the SGN segmentation is identified. To accomplish this, multiple methods will be added and compared based on their betweenness centrality.
